### DISCUSSION TITLE
Deepgram Speech to Text API - Websocket connection issue

### DISCUSSION THREAD

**Discussion by venkat-iscalepro** [deepgram/community#779] [General help]
### Which Deepgram product feature are you using?

Deepgram SDKs

### Details

We are working with the Deepgram Speech to Text API for generating transcriptions. We open a websocket, and once that is successful, we start our video recording. While this works well for most cases, we have found multiple occurrences where the ready state is being returned as 0 (Opening), as against 1(Open) or 2/3(Closing/Closed).

This has caused us significant trouble, as the only available events - onopen and onerror, neither of which trigger for ready state 0.

We had this for couple of days and it got automatically resolved, what is the root cause of the issue. How should we handle this scenario in the code.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

POST /v1/listen?model=nova-2

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

39006729-af12-486a-bb0c-bc97e50c4ba2

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.
